src/control/server/storage/bdev/backend.go

```diff
@@ -180,12 +169,24 @@ func convertControllers(bcs []spdk.Controller) ([]*storage.NvmeController, error
 	return scs, nil
 }
 
-func (b *spdkBackend) Scan() (storage.NvmeControllers, error) {
-	if err := b.Init(); err != nil {
+// Scan discovers NVMe controllers accessible by SPDK.
+func (b *spdkBackend) Scan(_ ScanRequest) (*ScanResponse, error) {
```

Comment:
Hmm. Does this mean that the scan will always find all devices irrespective of the white/blacklist values set in the config? I'm not sure if that's correct. I think if neither list is set then of course we should find everything, but otherwise I think we need to start honoring those values. Isn't this part of what [DAOS-4620](https://jira.hpdd.intel.com/browse/DAOS-4620) was about?

Reply:
we don't need a white list here because prepare has already applied the white/blacklists (bdev_{include,exclude} in the rebinding of the PCI devices, so only those filtered will be available to spdk